### ISSUE TITLE
Question/Help: Docker image runs locally but gives error in google cloud

### ISSUE THREAD

**Issue by xhulja07** [GoogleContainerTools/jib#2362]
Hello, I have a scala maven project and I am able to create the docker image using the `mvn clean compile jib:dockerBuild` command and run it locally. When I try pushing it to google cloud using `mvn -DskipTests com.google.cloud.tools:jib-maven-plugin:build -Dimage=gcr.io/myproject-ID/hello:v1` gives me the following error "" which looks like it doesn't recognize the scala library. 
When I pull the image from gcr it works fine.
Any help?  its the first time that I am using this plugin...

`Exception in thread "main" java.lang.NoClassDefFoundError: scala/Predef$
        at hello$.main(hello.scala:6)
        at hello.main(hello.scala)
Caused by: java.lang.ClassNotFoundException: scala.Predef$
        at java.net.URLClassLoader.findClass(URLClassLoader.java:382)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:419)
        at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:352)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:352)
        ... 2 more`

**Comment by chanseokoh** [comment]
Hi @xhulja07,

How do you declare the Scala library (presumably providing `scala/Predef` in your project? Is the library declared as a dependency in `pom.xml`? How does it look like?

And what's the output of running `mvn dependency:tree`?

**Comment by xhulja07** [comment]
Hi thanks for your help. Yes it is a dependency:

`[INFO] +- org.scala-lang:scala-library:jar:2.11.11:compile
[INFO] +- org.apache.ignite:ignite-indexing:jar:2.7.6:compile
[INFO] |  +- commons-codec:commons-codec:jar:1.11:compile
[INFO] |  +- org.apache.lucene:lucene-core:jar:7.4.0:compile
[INFO] |  +- org.apache.lucene:lucene-analyzers-common:jar:7.4.0:compile
[INFO] |  +- org.apache.lucene:lucene-queryparser:jar:7.4.0:compile
[INFO] |  |  +- org.apache.lucene:lucene-queries:jar:7.4.0:compile
[INFO] |  |  \- org.apache.lucene:lucene-sandbox:jar:7.4.0:compile
[INFO] |  \- com.h2database:h2:jar:1.4.197:compile
[INFO] +- org.apache.ignite:ignite-core:jar:2.7.6:compile
[INFO] |  +- javax.cache:cache-api:jar:1.0.0:compile
[INFO] |  +- org.jetbrains:annotations:jar:13.0:compile
[INFO] |  \- org.gridgain:ignite-shmem:jar:1.0.0:compile
[INFO] +- org.slf4j:slf4j-api:jar:1.7.5:compile
[INFO] +- org.apache.ignite:ignite-scalar_2.10:jar:2.7.6:compile
[INFO] +- org.apache.ignite:ignite-spring:jar:2.7.6:compile
[INFO] |  +- org.springframework:spring-core:jar:4.3.18.RELEASE:compile
[INFO] |  +- org.springframework:spring-aop:jar:4.3.18.RELEASE:compile
[INFO] |  +- org.springframework:spring-beans:jar:4.3.18.RELEASE:compile
[INFO] |  +- org.springframework:spring-context:jar:4.3.18.RELEASE:compile
[INFO] |  +- org.springframework:spring-expression:jar:4.3.18.RELEASE:compile
[INFO] |  +- org.springframework:spring-tx:jar:4.3.18.RELEASE:compile
[INFO] |  +- org.springframework:spring-jdbc:jar:4.3.18.RELEASE:compile
[INFO] |  \- commons-logging:commons-logging:jar:1.1.1:compile
[INFO] +- org.apache.ignite:ignite-ml:jar:2.7.6:compile
[INFO] |  +- it.unimi.dsi:fastutil:jar:7.0.12:compile
[INFO] |  +- com.github.fommil.netlib:core:jar:1.1.2:compile
[INFO] |  +- org.apache.commons:commons-rng-core:jar:1.0:compile
[INFO] |  |  \- org.apache.commons:commons-rng-client-api:jar:1.0:compile
[INFO] |  +- org.apache.commons:commons-rng-simple:jar:1.0:compile
[INFO] |  +- com.zaxxer:SparseBitSet:jar:1.0:compile
[INFO] |  \- org.apache.commons:commons-math3:jar:3.6.1:compile
[INFO] \- org.scalanlp:breeze_2.11:jar:0.10:compile
[INFO]    +- org.scalanlp:breeze-macros_2.11:jar:0.3.1:compile
[INFO]    |  \- org.scala-lang:scala-reflect:jar:2.11.0:compile
[INFO]    +- net.sourceforge.f2j:arpack_combined_all:jar:0.1:compile
[INFO]    +- net.sf.opencsv:opencsv:jar:2.3:compile
[INFO]    +- com.github.rwl:jtransforms:jar:2.4.0:compile
[INFO]    |  \- junit:junit:jar:4.8.2:compile
[INFO]    \- org.spire-math:spire_2.11:jar:0.7.4:compile
[INFO]       \- org.spire-math:spire-macros_2.11:jar:0.7.4:compile
`

**Comment by xhulja07** [comment]
I solved it... apparently I was using same tag for the image and the new image wasn't being pulled.

**Comment by soumiknow** [comment]
Recently I got the similar issue.

This time I am able to build the image using jib-maven-plugin from my local and able to deploy it into GCP Cloud Run. 

But when I used the same code to build the image using jib-maven-plugin from BitBucket pipeline, image got pushed to gcr, but unable to deploy to GCP CLoud Run. It says Error: Could not find or load main class com.demo.sdas.Application Caused by: java.lang.ClassNotFoundException: com.demo.sdas.Application.

What I am not understanding is, if this is working from my local then why it failed to load from BitBucket Pipeline. Everything is same in the code repository and executing the same commands for building the image and deploying it. Is there any bug in jib plugin.
